vital/types/transform.h

```diff
@@ -0,0 +1,81 @@
+/*ckwg +30
+ * Copyright 2019 by Kitware, Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *  * Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ *  * Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ *  * Neither name of Kitware, Inc. nor the names of any contributors may be
+ *    used to endorse or promote products derived from this software without
+ *    specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE AUTHORS OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ */
+
+/**
+ * \file
+ * \brief Core transform class definition
+ */
+
+#ifndef VITAL_TRANSFORM_H_
+#define VITAL_TRANSFORM_H_
+
+#include <vital/vital_export.h>
+#include <vital/vital_config.h>
+#include <vital/vital_types.h>
+
+#include <vital/types/vector.h>
+
+#include <memory>
+
+namespace kwiver {
+namespace vital {
+
+// Forward declarations of abstract transform class
+class transform;
+// typedef for a transform shared pointer
+typedef std::shared_ptr< transform > transform_sptr;
+
+// ============================================================================
+/// Abstract base transformation representation class
+class VITAL_EXPORT transform
```

Comment:
This is specifically a 2D transform.  We also have 3D transforms in KWIVER.  Should this be called `transform_2` and should we add a `transform_3` for `vector_3d`?  Possibly it should be `transform_2d`, but note that the `d` in `vector_2d` stands for `double`.  There is also `vector_2f`, `vector_2i`, etc. but the higher level classes almost exclusively use the `double` variants.

Reply:
Possibly, though as you seem to notice, what to name it isn't obvious. I was inclined to punt on that until it matters, and rename later if needed.

Reply:
Okay, per F2F discussion, the class is now `transform_2d`. Note that the `d` does actually refer to `double`, in that `map` operates on vectors whose scalar type is `double`.